Bottom: 551bec14dd6590a208a351b3046f8b63735bba25
Top:    347061dffbe9d2f971e7d3896736ae594ff3614c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 02:36:03 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index e16fd55..561bda2 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -253,6 +253,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
